src/Select/SelectInput.js

```diff
@@ -345,6 +345,7 @@ class SelectInput extends React.Component<ProvidedProps & Props, State> {
           onKeyDown={this.handleKeyDown}
           onBlur={this.handleBlur}
           onClick={disabled || readOnly ? null : this.handleClick}
```

Comment:
Do we still need this one?

Reply:
Originally I thought we needed it for handling the keyboard events, but just realized that this is already explicitly handled by the onKeyDown event. Also tested without the onClick on mobile and desktop, and seems to still work without it.

I fixed this in the second commit.

Reply:
Forgot to fix the unit tests though. This got me thinking that maybe we should keep the second handler for testing's sake. What do you think?

Reply:
We can update the tests too. The tests needs to be as close as what people use in the browser.